portal/views/assessment_engine.py

```diff
@@ -1623,6 +1618,11 @@ def present_needed():
         return redirect('/')
 
     url = url_for('.present_assessment', **args)
+
+    encounter = current_user().current_encounter
+    if encounter.auth_method == "url_authenticated":
+        return redirect('/confirm-identity?redirect_url={}'.format(url))
```

Comment:
I've found redirects really hard to debug, would it make sense to add a logging line here before redirecting?

Reply:
thanks, ivan, added